sklego/pandas_utils.py

```diff
@@ -0,0 +1,152 @@
+import inspect
+import logging
+import numpy as np
+import pandas as pd
+import datetime as dt
+
+from functools import wraps
+from sklego.common import as_list
+from scipy.ndimage.interpolation import shift
+
+
+def log_step(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        logger = logging.getLogger(inspect.stack()[1].function)
+
+        tic = dt.datetime.now()
+        result_df = func(*args, **kwargs)
+        time_taken = str(dt.datetime.now() - tic)
+        func_args = inspect.signature(func).bind(*args, **kwargs).arguments
+        func_args_str = ''.join(', {} = {!r}'.format(*item) for item in list(func_args.items())[1:])
+
+        logger.info(
+            f"[ {func.__name__}(df{func_args_str}) ] "
+            f"n_obs={result_df.shape[0]} n_col={result_df.shape[1]} time={time_taken}")
+        return result_df
+    return wrapper
+
+
+def add_lags(X, cols, lags):
```

Comment:
two final things (we're getting close tho!) 

1. `add_lags` isn't tested at the moment. a few tests that confirm error messages would be nice. 
2. @MBrouns we might want to add base behavior to allow the `NA` rows to still be kept around.

Reply:
A drop_na flag that is `True` by default?

Reply:
sounds good.

Reply:
one that is in and tested ill click the green merge button.